modules/reindex/src/main/java/org/elasticsearch/reindex/TransportDeleteByQueryAction.java

```diff
@@ -64,8 +67,17 @@ public void doExecute(Task task, DeleteByQueryRequest request, ActionListener<Bu
                     clusterService.localNode(),
                     bulkByScrollTask
                 );
-                new AsyncDeleteByQueryAction(bulkByScrollTask, logger, assigningClient, threadPool, request, scriptService, listener)
-                    .start();
+                new AsyncDeleteByQueryAction(
+                    bulkByScrollTask,
+                    logger,
+                    assigningClient,
+                    threadPool,
+                    request,
+                    scriptService,
+                    ActionListener.runAfter(listener, () -> {
+                        long elapsedTime = TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime);
```

Comment:
looks like this variable is unused?

Reply:
Ya I need to capture this into a metrics. Sorry this was still WIP.